docs: updates (#1587)
terramate-io/terramate: docs/cli/orchestration/run-commands-in-stacks.md

Terramate integrates with various tools to enable [Change Detection](../change-detection/index.md).

Making use of Change Detection features when running commands can improve run-times on local machines and in automation.

To enable it add the `--changed` command line option.

**Example:** Execute a command in all stacks that contain changes

```hcl
terramate run --changed -- echo "hello from changed stack"
```

## Influence the order of execution

Terramate honors the explicit and implicit order of execution when running commands.

`terramate list --run-order` or `terramate run --dry-run` can be used to preview the order in which commands will be executed in stacks.

### Run in parallel

Stacks that are not affected by a specific order of execution can be executed in parallel.

Terramate will always guarantee that ordered stacks will still run in order but independent stacks or stacks that have their depending stacks completed can run in parallel.

By default, Terramate will always execute all stacks in sequence one stack at a time.

**Example:** Run multiple stacks in parallel

```hcl
terramate run --parallel 100 -- echo "hello from stack in parallel"
```

::: warning

> It is not possible to run `terraform init` in parallel when provider caching is enabled via `TF_PLUGIN_CACHE_DIR` as terraform does not support this mode of operation at this time.

Reviewer: ```suggestion
It is not possible to run `terraform init` in parallel when provider caching is enabled via `TF_PLUGIN_CACHE_DIR` as Terraform does not support this mode of operation at this time.
```

:::

### Reverse the run order

The order defined in the configuration of a stack and defined via integrations can be reverted when executing commands in stacks.

**Example:** Execute a command in all stacks but in reverse order

```hcl
terramate run --reverse -- echo "hello from stack in reversed order"
```

This is useful when running destructive operations where dependent stacks need to remove their configuration before other stacks.

> A very good example here is the destroy operation of Terraform to destroy stacks in opposite order:

Reviewer: ```suggestion
An example use-case is the destroy operation of Terraform to destroy stacks in opposite order:
```
